packages/core/src/processed-data-manager.ts

```diff
@@ -0,0 +1,134 @@
+import get from 'lodash/get';
+import { ApplicationError } from 'n8n-workflow';
+import type {
+	IProcessedDataManager,
+	ICheckProcessedOptions,
+	ICheckProcessedOutput,
+	ICheckProcessedOutputItems,
+	IDataObject,
+	ProcessedDataContext,
+	ProcessedDataItemTypes,
+	ICheckProcessedContextData,
+} from 'n8n-workflow';
+
+export class ProcessedDataManager {
+	private static instance: ProcessedDataManager;
+
+	private manager: IProcessedDataManager;
+
+	private constructor(manager: IProcessedDataManager) {
+		this.manager = manager;
+	}
+
+	static async init(manager: IProcessedDataManager): Promise<void> {
+		if (ProcessedDataManager.instance) {
+			throw new ApplicationError('Processed Data Manager is already initialized');
+		}
+
+		ProcessedDataManager.instance = new ProcessedDataManager(manager);
+	}
+
+	static getInstance(): ProcessedDataManager {
+		if (!ProcessedDataManager.instance) {
+			throw new ApplicationError('Processed Data Manager is not initialized');
+		}
+		return ProcessedDataManager.instance;
+	}
+
+	static getManager(): string[] {
+		return Object.keys(ProcessedDataManager.instance.manager);
+	}
+
+	async checkProcessed(
+		items: ProcessedDataItemTypes[],
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+		options: ICheckProcessedOptions,
+	): Promise<ICheckProcessedOutput> {
+		if (this.manager) {
+			return await this.manager.checkProcessed(items, context, contextData, options);
+		}
+		throw new ApplicationError('There is no manager');
+	}
+
+	async checkProcessedItemsAndRecord(
+		propertyName: string,
+		items: IDataObject[],
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+		options: ICheckProcessedOptions,
+	): Promise<ICheckProcessedOutputItems> {
+		if (!this.manager) {
+			throw new ApplicationError('There is no manager');
+		}
```

Comment:
This is a recurring pattern. Could we move it to private method? something like:

```ts
private assertManager() {
  a.ok(this.manager, "Manager needs to initialized before use. Make sure to call init()");
}
```

where `a` is `import * as a from "node:assert/strict"`